lib/postcode_validation/use_case/address_list.rb

```diff
@@ -0,0 +1,66 @@
+require_relative 'validate_address/format_validator'
+
+module PostcodeValidation
+  module UseCase
+    class AddressList
+      def initialize(address_list_gateway:, logger: nil)
+        @format_validator = PostcodeValidation::UseCase::ValidateAddress::FormatValidator.new
+        @address_list_gateway = address_list_gateway
+        @logger = logger
+        @errors = []
+      end
+
+      def execute(postcode:, country:)
+        check_country(country)
+        check_postcode_format(postcode, country)
+        result = matched_addresses(postcode, country)
```

Comment:
This is pretty much a carbon copy of the other usecase.  Also not entirely sure how to get rid of the duplication.